### PR TITLE
GRF-376: Filter categories on search filter

### PR DESCRIPTION
**Pull request by etiennekerbastard**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php [hunk 1]
```diff
@@ -134,7 +134,7 @@ protected function addFilters(QueryBuilder $qb, array $searchFilters): QueryBuil
                 }
             }
         }
-
+        $toto = $qb->getQuery()->getSQL();
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php:137]: sacré toto

File: src/Akeneo/Category/back/Infrastructure/DTO/ExternalApiSqlParameters.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Infrastructure\DTO;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ExternalApiSqlParameters
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/back/Infrastructure/DTO/ExternalApiSqlParameters.php:11]: you can put this file directly in Application/Query

File: src/Akeneo/Category/back/ServiceApi/ExternalApiCategory.php [hunk 3]
```diff
@@ -115,14 +115,14 @@ public function normalize(bool $withPosition, bool $withEnrichedAttributes): arr
...
             'labels' => $this->getLabels(),
-            'template' => $this->getTemplateCode(),
         ];
 
         if ($withPosition) {
             $normalizedCategory['position'] = $this->getPosition();
         }
 
         if ($withEnrichedAttributes) {
+            $normalizedCategory['template'] = $this->getTemplateCode();
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/ServiceApi/ExternalApiCategory.php:125]: Il faudrait ajouterle test aussi

Comment (etiennekerbastard) [reply src/Akeneo/Category/back/ServiceApi/ExternalApiCategory.php:125]: I will create a ticket for this as this is out of the scope of this PR

**Review by jeremyBeaucousinAkeneo — APPROVED** [review]

**Review by BryanS29-akeneo — APPROVED** [review]

**Pull request merged**